### PR TITLE
When parsing project.toml, don't warn about unexpected keys that are actually expected

### PR DESCRIPTION
**Pull request by natalieparellano**
Fixes https://github.com/buildpacks/pack/issues/2192

## Documentation

- Should this change be documented?
    - [ ] Yes, see #___
    - [ ] No

## Related

Resolves #___

### CODE REVIEW

File: pkg/project/types/types.go [hunk 1]
```diff
@@ -33,10 +33,13 @@ type Build struct {
...
-	Version   string    `toml:"version"`
-	SourceURL string    `toml:"source-url"`
-	Licenses  []License `toml:"licenses"`
+	ID               string    `toml:"id"`
+	Name             string    `toml:"name"`
+	Version          string    `toml:"version"`
+	Authors          []string  `toml:"authors"`
+	DocumentationURL string    `toml:"documentation-url"`
+	SourceURL        string    `toml:"source-url"`
+	Licenses         []License `toml:"licenses"`
```
Comment (natalieparellano) [inline pkg/project/types/types.go:42]: According to https://github.com/buildpacks/spec/blob/extensions/project-descriptor/0.1/extensions/project-descriptor.md#project

File: pkg/project/v02/project.go [hunk 2]
```diff
@@ -28,10 +28,15 @@ type Env struct {
...
-	SchemaVersion string                 `toml:"schema-version"`
+	SchemaVersion    string                 `toml:"schema-version"`
+	ID               string                 `toml:"id"`
+	Name             string                 `toml:"name"`
+	Version          string                 `toml:"version"`
+	Authors          []string               `toml:"authors"`
+	Licenses         []types.License        `toml:"licenses"`
+	DocumentationURL string                 `toml:"documentation-url"`
+	SourceURL        string                 `toml:"source-url"`
+	Metadata         map[string]interface{} `toml:"metadata"`
```
Comment (natalieparellano) [inline pkg/project/v02/project.go:39]: According to https://github.com/buildpacks/spec/blob/extensions/project-descriptor/0.2/extensions/project-descriptor.md#_

**Comment by natalieparellano** [comment]
@jjbustamante would you mind taking a look at this one?

**Review by jjbustamante — APPROVED** [review]
Thanks @natalieparellano!

**Pull request merged**